src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationProviders.xml

```diff
@@ -2153,6 +2153,27 @@
     <Environment Issuer="https://api.login.yahoo.com/" />
   </Provider>
 
+  <Provider Name="Zendesk" Id="5577b117-1f3d-464f-b987-aabf7c0ab3f7" Documentation="https://developer.zendesk.com/api-reference/introduction/security-and-auth/#oauth-access-token">
+
+    <!--Note: Zendesk is a multitenant provider that relies on subdomains to identify instances.
+      As such, the following URLs all include a {settings.Tenant} placeholder that will be dynamically
+      replaced by OpenIddict at runtime by the tenant configured in the Zendesk settings.-->
+
+    <Environment Issuer="https://{settings.Tenant}.zendesk.com/">
+      <Configuration AuthorizationEndpoint="https://{settings.Tenant}.zendesk.com/oauth/authorizations/new"
+                     TokenEndpoint="https://{settings.Tenant}.zendesk.com/oauth/tokens"
+                     UserInfoEndpoint="https://{settings.Tenant}.zendesk.com/api/v2/users/me">
+        <CodeChallengeMethod Value="S256" />
+        <GrantType Value="authorization_code" />
+
+      </Configuration>
+    </Environment>
+    <Setting PropertyName="Tenant" ParameterName="tenant" Type="String" Required="true"
+             Description="Gets or sets the tenant used to identify the Zendesk instance." />
+    <Setting PropertyName="ClientType" ParameterName="kind" Type="String" Required="false"
+         DefaultValue="confidential" Description="Client Type: public or confidential. Public OAuth clients are applications that run in environments where credentials cannot be securely stored, such as mobile and web apps. These clients are required to use PKCE. Confidential OAuth clients run on secure servers where their credentials can be kept secure. These clients can use PKCE, client secret, or both." />
```

Comment:
```suggestion
```

This setting shouldn't be needed (and since you didn't write custom code to do something based on its value, it really has no effect whatsoever) 😃

Reply:
> Should the value not be automatically added to the code request as it has both a property name and a parameter name?

Nope (note: the "parameter name" here refers to the name of the parameter used in the C# builder method, not an OIDC request parameter).

> if not where should I update to include this value as this is required or the client secret is required. I would like to also add validation for configuration.

You don't need that parameter: when no client secret or signing credentials is attached to a client registration, OpenIddict detects it when negotiating the client authentication method and sends the token request without a secret attached.

There's no need for special validation logic for Zendesk, it behaves like any other standard provider in this regard.